### PR TITLE
Added a tutorial for progress updates

### PR DESCRIPTION
**Pull request by ACPixel**
Felt we needed one of these, so i asked probable about it. he said sure, so hey. here it is :)

### CODE REVIEW

File: src/tutorials/code/node/interactive/progress/2.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+let json = {"tactile": [new Packets.ProgressUpdate.TactileUpdate({"id": 0, "cooldown": 30000})]};
+let update = Packets.ProgressUpdate(json);
+robot.send(Packets.ProgressUpdate(update));
```
Comment (UnwrittenFun) [inline src/tutorials/code/node/interactive/progress/2.js]: Wrapping it in `Packets.ProgressUpdate` twice? Is that correct?

Comment (ProbablePrime) [inline src/tutorials/code/node/interactive/progress/2.js]: No, this is wrong. 

```
const update = new Packets.ProgressUpdate(json);
robot.send(update);
```
Would be correct, @ACPixel the new is important. We had this discussion in gitter yesterday :'(

File: src/tutorials/code/node/interactive/progress/2.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+let json = {"tactile": [new Packets.ProgressUpdate.TactileUpdate({"id": 0, "cooldown": 30000})]};
```
Comment (ProbablePrime) [inline src/tutorials/code/node/interactive/progress/2.js]: this doesn't need to be json:
```
{
  tactile: [
    new Packets.ProgressUpdate.TactileUpdate({id: 0, cooldown: 30000})
  ]
};

Looks better for clarity

File: src/tutorials/pupdate.pug [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    h2#controlEditor Types of Progress Updates.
+    p.
+        There are various types of progress updates, or rather progress updates
+        for different things. Such as Joystick updates of tactile updates.
+        You can also do screen updates if you are using screen controls.
+        Below will be a list of progress update tags, for more info go to the
+        #[a(href='/reference/interactive/index.html') reference guide].
+
+    table.table
```
Comment (ProbablePrime) [inline src/tutorials/pupdate.pug]: We'll be automatically generating this table as a part of the reference material at some point

**Comment by ProbablePrime** [comment]
While awesome, this tutorial would be confusing.

The first tutorial deals with just joysticks, tactile code in there wouldn't make much sense or achieve much.

A better plan for this tutorial would be:

- Talk the user through adding a button with analysis types to the developer lab
- using the code from the first tutorial(link to it sure but maybe attach it as a snippet anyway, I need to work on making this simpler, our bootstrap should be easier)
- Processing a report showing a push on that button and triggering a cooldown.

File: src/tutorials/pupdate.pug (hunk 3 above)
Comment (SimonSchick) [inline src/tutorials/pupdate.pug]: I guess this would be a good opportunity to do so, since this is a relatively big chunk of this PR.

Comment (ProbablePrime) [inline src/tutorials/pupdate.pug]: Its been on my list for quite awhile. I'll find some time at some point, hopefully today.

File: src/tutorials/code/node/interactive/progress/2.js (hunk 2 above)
Comment (SimonSchick) [inline src/tutorials/code/node/interactive/progress/2.js]: While `let` works, `const` would be better.

File: src/tutorials/code/node/interactive/progress/2.js (hunk 1 above)
Comment (ACPixel) [inline src/tutorials/code/node/interactive/progress/2.js]: i am so sorry, i totally thought i copied the right code... ;(

**Comment by ACPixel** [comment]
@ProbablePrime i will do all of that when i have the time, sorry for not doing it to its fullest, I was rushing a bit

**Pull request closed without merging**